src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Tests.cs

```diff
@@ -78,6 +78,54 @@ public void PriorityQueue_Generic_EnqueueDequeue_EqualToMin()
             Assert.True(enqueuedItems.SetEquals(queue.UnorderedItems));
         }
 
+        [Fact]
+        public void PriorityQueue_EmptyCollection_DequeueEnqueue_ShouldThrowInvalidOperationException()
+        {
+            var queue = new PriorityQueue<int, int>();
+
+            Assert.Equal(0, queue.Count);
+            Assert.Throws<InvalidOperationException>(() => queue.DequeueEnqueue(1, 1));
```

Comment:
Perhaps add an assertion validating there is no state change
```suggestion
            Assert.Throws<InvalidOperationException>(() => queue.DequeueEnqueue(1, 1));
            Assert.Equal(0, queue.Count);
```

Reply:
Added the additional suggested assertion.